src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,214 @@
+      rightMotor.setControl(motionMagicRequest.withVelocity(desiredRightVelocity.in(Units.RotationsPerSecond)));
+    }
+  }
+
+  public void setLeftShooterIntakeVoltage(double voltage) {
+    leftMotor.setControl(voltageRequest.withOutput(voltage));
+
+  }
+
+  public void setRightShooterIntakeVoltage(double voltage) {
+    rightMotor.setControl(voltageRequest.withOutput(voltage));
+  }
+
+  /**
+   * Sets all of the shooting motors to neutral.
+   */
+  public void setShootingNeutralOutput() {
+    leftMotor.setControl(new NeutralOut());
+    rightMotor.setControl(new NeutralOut());
+  }
+
+  /**
+   * @return The current velocity of the left shooter motor. <b> Units: </b>
+   *         Rotations per second
+   */
+  public double getLeftShooterVelocity() {
+    return leftMotor.getVelocity().getValueAsDouble();
+  }
+
+  /**
+   * @return The current velocity of the right shooter motor. <b> Units: </b>
+   *         Rotations per second
+   */
+  public double getRightShooterVelocity() {
+    return rightMotor.getVelocity().getValueAsDouble();
+  }
+
+  /**
+   * @return If the left shooter motor is at its desired velocity
+   */
+  public boolean isLeftShooterUpToSpeed() {
+    return (Math.abs(
+        getLeftShooterVelocity()
+            - desiredLeftVelocity.in(Units.RotationsPerSecond))) <= constShooter.UP_TO_SPEED_TOLERANCE
+                .in(Units.RotationsPerSecond);
+  }
+
+  /**
+   * @return If the right shooter motor is at its desired velocity
+   */
+  public boolean isRightShooterUpToSpeed() {
+    return (Math.abs(getRightShooterVelocity()
+        - desiredRightVelocity.in(Units.RotationsPerSecond))) <= constShooter.UP_TO_SPEED_TOLERANCE
+            .in(Units.RotationsPerSecond);
+  }
+
+  /**
+   * @return If both motors have a non-zero desired velocity and are at their
+   *         desired velocities
+   */
+  public boolean areBothShootersUpToSpeed() {
+    return (isLeftShooterUpToSpeed()
+        && isRightShooterUpToSpeed() && (getLeftShooterVelocity() != 0 || getRightShooterVelocity() != 0))
+        || ignoreFlywheelSpeed;
+  }
+
+  public void setLeftDesiredVelocity(Measure<Velocity<Angle>> desiredVelocity) {
+    desiredLeftVelocity = desiredVelocity;
+  }
+
+  public void setRightDesiredVelocity(Measure<Velocity<Angle>> desiredVelocity) {
+    desiredRightVelocity = desiredVelocity;
+  }
+
+  public void setDesiredVelocities(Measure<Velocity<Angle>> desiredLeftVelocity,
+      Measure<Velocity<Angle>> desiredRightVelocity) {
+    setLeftDesiredVelocity(desiredLeftVelocity);
+    setRightDesiredVelocity(desiredRightVelocity);
+  }
+
+  public void setVoltage(double leftVoltage, double rightVoltage) {
+    setLeftShooterIntakeVoltage(leftVoltage);
+    setRightShooterIntakeVoltage(rightVoltage);
+  }
+
+  public void setIgnoreFlywheelSpeed(boolean ignoreFlywheelSpeed) {
+    this.ignoreFlywheelSpeed = ignoreFlywheelSpeed;
+  }
+
+  public void setShooterPosition(Measure<Angle> position) {
```

Comment:
```suggestion
  public void setShooterAngle(Measure<Angle> position) {
```
I think calling it shooter angle is a bit more clear. when first reading this I thought this set position of the flywheel (like fly wheel angle). I am ok with leaving as is though.